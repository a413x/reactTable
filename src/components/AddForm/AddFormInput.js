import {validate} from '../../utils/validate.js'

export const AddFormInput = ({
  colName, addingChange, validChange, valid
}) => {
  const onChange = (value) => {
    let valueValid = false
    if(colName === 'id') {
      valueValid = validate('number', value)
    }else if(colName === 'firstName' || colName === 'lastName'){
      valueValid = validate('text', value)
    }else if(colName === 'email'){
      valueValid = validate('email', value)
    }else if(colName === 'phone'){
      valueValid = validate('phone', value)
    }
    validChange(colName, valueValid)
    addingChange(colName, value)
  }

  const validClass = valid ? 'is-valid' : 'is-invalid'
  return (
    <div>
      <label>{colName}</label>
      <input
        className = {'form-control ' + validClass}
        onChange = {(e) => onChange(e.target.value)}
      />
      <div className = 'invalid-feedback'>
        Please enter correct value
      </div>
    </div>
  )
}
