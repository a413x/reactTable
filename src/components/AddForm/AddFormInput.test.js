import {render, fireEvent} from '@testing-library/react'
import {AddFormInput} from './AddFormInput.js'

describe('AddFormInput component tests', () => {
  const colName = 'id'
  const mock_addingChange = jest.fn()
  const mock_validChange = jest.fn()
  const valid = false

  const props = {
    colName: colName,
    addingChange: mock_addingChange,
    validChange: mock_validChange,
    valid: valid
  }

  test('renders properly', () => {
    const container = render(<AddFormInput {...props} />)
    expect(container).toMatchSnapshot()
  })

  test('renders properly with true valid prop', () => {
    props.valid = true
    const container = render(<AddFormInput {...props} />)
    expect(container).toMatchSnapshot()
  })

  test('input with correct value calls callback functions', () => {
    const value = '1'
    render(<AddFormInput {...props} />)
    fireEvent.change(document.querySelector('input'), {
      target: {value: value }
    })
    expect(mock_validChange).toHaveBeenCalledWith(colName, true)
    expect(mock_addingChange).toHaveBeenCalledWith(colName, value)
  })

  test('input with wrong value calls callback functions', () => {
    const value = 'some text'
    render(<AddFormInput {...props} />)
    fireEvent.change(document.querySelector('input'), {
      target: {value: value }
    })
    expect(mock_validChange).toHaveBeenCalledWith(colName, false)
    expect(mock_addingChange).toHaveBeenCalledWith(colName, value)
  })
})
