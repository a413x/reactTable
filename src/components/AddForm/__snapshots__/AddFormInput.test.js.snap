// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFormInput component tests renders properly 1`] = `
<div>
  <div>
    <label>
      id
    </label>
    <input
      class="form-control is-invalid"
    />
    <div
      class="invalid-feedback"
    >
      Please enter correct value
    </div>
  </div>
</div>
`;

exports[`AddFormInput component tests renders properly with true valid prop 1`] = `
<div>
  <div>
    <label>
      id
    </label>
    <input
      class="form-control is-valid"
    />
    <div
      class="invalid-feedback"
    >
      Please enter correct value
    </div>
  </div>
</div>
`;
